### PR TITLE
Move VIP to lua

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Move VIP to lua.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: data/lib/core/core.lua [hunk 1]
```diff
@@ -19,4 +19,5 @@ dofile('data/lib/core/position.lua')
 dofile('data/lib/core/teleport.lua')
 dofile('data/lib/core/tile.lua')
 dofile('data/lib/core/vocation.lua')
+dofile('data/lib/core/vip.lua')
```
Comment (ranisalt) [inline data/lib/core/core.lua:22]: Alphabetical order

File: data/lib/core/player.lua [hunk 2]
```diff
@@ -698,3 +698,140 @@ end
...
+		self:sendTextMessage(MESSAGE_STATUS_SMALL, "This player is already in your list.")
+		return
+	end
+
+	local description = ""
+	local icon = 0
+	local notify = false
+
+	local accountId = self:getAccountId()
+	db.asyncQuery("INSERT INTO `account_viplist` (`account_id`, `player_id`, `description`, `icon`, `notify`) VALUES (" .. accountId .. ", " .. vipGuid .. ", " .. db.escapeString(description) .. ", " .. icon .. ", " .. (notify and 1 or 0) .. ")")
```
Comment (ranisalt) [inline data/lib/core/player.lua:771]: These are the defaults

```suggestion
	local accountId = self:getAccountId()
	db.asyncQuery("INSERT INTO `account_viplist` (`account_id`, `player_id`) VALUES (" .. accountId .. ", " .. vipGuid .. ")")
```

File: data/lib/core/player.lua [hunk 3]
```diff
@@ -698,3 +698,140 @@ end
...
+end
+
+function Player.sendVIP(self, guid, name, description, icon, notify, status)
+	local msg = NetworkMessage()
+	msg:addByte(0xD2)
+	msg:addU32(guid)
+	msg:addString(name)
+	msg:addString(description)
+	msg:addU32(math.min(10, icon))
+	msg:addByte(notify and 0x01 or 0x00)
```
Comment (ranisalt) [inline data/lib/core/player.lua:821]:
```suggestion
	msg:addByte(notify and 1 or 0)
```

**Comment by moviebr** [comment]
I liked this code, it was well done. Good code.

File: src/luascript.cpp [hunk 4]
```diff
@@ -10710,14 +10712,14 @@
...
-				it.second->notifyStatusChange(player, VIPSTATUS_OFFLINE);
+				//it.second->notifyStatusChange(player, VIPSTATUS_OFFLINE);
 			}
 		}
 		IOLoginData::updateOnlineStatus(player->getGUID(), false);
 	} else {
 		for (const auto& it : g_game.getPlayers()) {
 			if (!it.second->isAccessPlayer()) {
-				it.second->notifyStatusChange(player, VIPSTATUS_ONLINE);
+				//it.second->notifyStatusChange(player, VIPSTATUS_ONLINE);
```
[the remark this answers is not in the document]
Comment (ramon-bernardo) [reply src/luascript.cpp:10722]: Bot, i need #4586 to move this

File: src/luascript.cpp [hunk 5]
```diff
@@ -10710,14 +10712,14 @@
 	if (player->isInGhostMode()) {
 		for (const auto& it : g_game.getPlayers()) {
 			if (!it.second->isAccessPlayer()) {
-				it.second->notifyStatusChange(player, VIPSTATUS_OFFLINE);
+				//it.second->notifyStatusChange(player, VIPSTATUS_OFFLINE);
```
[the remark this answers is not in the document]
Comment (ramon-bernardo) [reply src/luascript.cpp]: add an alternative by millhiore

**Pull request closed without merging**